packages/teams-js/src/private/associatedApps.ts

```diff
@@ -0,0 +1,237 @@
+import { sendMessageToParentAsync } from '../internal/communication';
+import { ensureInitialized } from '../internal/internalAPIs';
+import { ErrorCode, SdkError, TabInstance } from '../public';
+import { runtime } from '../public/runtime';
+
+// Open questions
+// 1. I've re-used `TabInstance` from the public API, does that contain all of the information you and app developers might need?
+// 2. I didn't see any reason to add a `getTabs` function because `pages.tabs.getTabInstances`. Any reason that won't work for you?
+// 3. I've added an `AppTypes[]` param to `addAndConfigureApp` to allow for the host to show different app types to the user. Very open to changes.
+// 4. I've added empty, private `validate` functions for the threadId and TabInstance. Any validation that is possible will help prevent against
+//    bad data being sent to the host. If you have any validation that can be done, please add it there. If you *can* use restrictive types like UUID
+//    or something, that would be even better.
+// 5. I've made the namespace structure an empty `associatedApps` namespace that only contains the `tab` namespace. This was an attempt to leave room for
+//    expansion in the future for non-tab scenarios that will make it less likely that your callers will need to update their code. Open to opinions though.
+
+// TODO: Add unit tests
+// TODO: Add E2E tests
+
+/**
+ * @hidden
+ * @internal
+ * @beta
+ * Limited to Microsoft-internal use
+ *
+ * TODO: Brief description of what this capability does
+ */
+export namespace associatedApps {
```

Comment:
I am still a bit iffy with the 'associatedApps' name. I know that we dont want to use extensibilityApps since 'extensibility' is a Teams concept.

But 'platformApps' might be better. Since 'platform' should be generic enough for all hubs where developers want to build their apps on the platform

Reply:
I am also not wild about the associatedApps name, and happy to get more opinions. I'm also not wild about the "platformApps" name though, because I don't know what "platform apps" are. Aren't Teams apps platform apps? What makes a platform app different than a non-platform app?